### ISSUE TITLE
Attendees cannot raiseHand

### ISSUE THREAD

**Issue by scredii** [aws-samples/amazon-chime-live-events#33]
When attendees click on "Raise Hand" we have a socket error:

```
index.tsx:143 Error sending Live Event message.
index.tsx:171 DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.
```

Attendees and moderators are connected to the same socket URL:

```
wss://xxxx.execute-api.us-east-1.amazonaws.com/Prod?LiveEventId=LiveEvent-xxxx&AttendeeId=
wss://xxxx.execute-api.us-east-1.amazonaws.com/Prod?LiveEventId=LiveEvent-xxxx&AttendeeId=
```
Any solution ?

thank you

**Comment by michhyun1** [comment]
Hi @scredii, thanks for bringing this up - we found this bug on our end yesterday also. I'll take a look and see if I can find out more. Thanks!

**Comment by scredii** [comment]
Hi @michhyun1.
Thanks for your quick reply. 
Do you have any update ?
thanks

**Comment by vijayamin83** [comment]
Is there any update on this?

**Comment by michhyun1** [comment]
@scredii @vijayamin83 sorry for the delay, I need to take a deeper look into this, I'll try to post an update by the end of the week if that is okay with you.

**Comment by msgoo7** [comment]
Hi, 
Facing same issue, Attendees not able to join also and not able to raise hand. Can you please update.

**Comment by michhyun1** [comment]
@scredii @vijayamin83 @msgoo7 thank you for your patience. I've been able to reproduce this, I see the following:

![image](https://user-images.githubusercontent.com/54328451/118732790-dabfc600-b7ef-11eb-8e2d-51917e085cb2.png)

I do notice that the console log says that the attendee is not authenticated. Can you confirm this is the same for you?

**Comment by msgoo7** [comment]
Hi @michhyun1 ,Thanks for your response.
I am getting the same error. 
There is no issues in broadcast, moderator and talent. But when we login to attendee page via the url from s3 bucket, got a similar response which you attached in the image.
Due to which when attendees raised hand , no notification is there to the moderator and also when the moderator broadcast the live session, attendees is also not able to see the same and a constant screen saying no one is streaming appears on the attendee page.

**Comment by Ashrma2021** [comment]
I am getting below error while trying to raise hand.

![image](https://user-images.githubusercontent.com/82191937/118952378-86485200-b979-11eb-9930-238bb6d09e23.png)

**Comment by vijayamin83** [comment]
@michhyun1 it show attendee unauthenticated, even if I use the authenticated attendee for that particular events. 
Where as other users like moderator, talent and broadcaster. 
I have cross verified in the dynamo table also and it contain the valid data. But still getting an error.

**Comment by michhyun1** [comment]
@vijayamin83 @Ashrma2021 @msgoo7 @scredii 

Thanks all for your patience - it seems like there was a bug introduced when we updated our dependency to the SDK to 2.0.0.

I've gone ahead and deployed a version using the git commit hash (30c8f9e3ad66825a7088e2f4843cebdf33c51ca4) - This is the commit right before the update to the 2.0.0 version of the SDK. You may want to do the same, as the raise hand feature will work on this commit. 

Sorry for the delay - hope this unblocks you for the time being until we can make the necessary fixes. Thanks.

**Comment by Ashrma2021** [comment]
Hi, We are still having issue. This time error is different.

![image](https://user-images.githubusercontent.com/82191937/119091250-67ee5f00-ba2a-11eb-8eb0-617bcd352dfd.png)

**Comment by michhyun1** [comment]
@Ashrma2021 Please check that you've create a completely new stack and bucket before trying again. Which commit are you using?

![image](https://user-images.githubusercontent.com/54328451/119175672-f78e1080-ba1e-11eb-8ae8-45eb12521399.png)


I'm able to see the raised hand in the moderator's view using the https://github.com/aws-samples/amazon-chime-live-events/commit/30c8f9e3ad66825a7088e2f4843cebdf33c51ca4 commit.

**Comment by michhyun1** [comment]
Also, You should make sure to enable the CORs extension.

**Comment by vijayamin83** [comment]
I have tried attendee view with downgrading chime sdk as "30c8f9e", but still I am not able see the attendee view.

<img width="847" alt="image" src="https://user-images.githubusercontent.com/468540/119943552-cf6b5800-bfb0-11eb-847a-39803d0d1038.png">

**Comment by michhyun1** [comment]
@vijayamin83 it seems as if the attendee URL works, but there is no video feed. In order to enable the video feed on the Attendee view, you need to follow a few more steps. 

Make sure you have completed all of the steps [here](https://github.com/aws-samples/amazon-chime-live-events/tree/master/src/backend/serverless#instructions).

One more thing you may need to do is follow [this guide](https://github.com/aws-samples/amazon-chime-live-events/tree/master/src/backend/serverless#transcoder-serverless-stack). You will need to start transcoding like [this](https://github.com/aws-samples/amazon-chime-live-events/blob/master/transcoding/README.md#options-to-startstop-transcoding).

Once you have done these steps, join as a Moderator. Enable your video and go live as a Moderator. Then, Join as an attendee and you should see the live streaming of the Moderator's video.

**Comment by vijayamin83** [comment]
@michhyun1 I have followed the all steps as you mentioned above. Also started the channel ID in MediaLive before broad casting. But still, I am not able to get the attendee video working.

When I check the related MediaPackage HLS stream it also not working. Please see the screen shot.

<img width="881" alt="image" src="https://user-images.githubusercontent.com/468540/120589459-8145d100-c456-11eb-8550-c0bd72cafe75.png">

**Comment by vijayamin83** [comment]
I have also checked the "30c8f9e" commit. But still getting the error.

<img width="1187" alt="image" src="https://user-images.githubusercontent.com/468540/120632091-39d73900-c486-11eb-8175-01e9d6fa5026.png">

**Comment by michhyun1** [comment]
You may want to double check the transcoding start lambda test event. You'll need to stop the transcoding and then start it again with the test event being:
```
{
  "action": "start",
  "meetingURL": "<your-meeting-url-to-be-broadcast>", <-- use the broadcast meeting URL
  "rtmpEndpoint": "<your-rtmp-endpoint-url>" <-- this should be part of your outputs
}
```

**Comment by vijayamin83** [comment]
Ok I am able to run this broadcast lambda function, But stilll I am not getting the attendee view's video.

Also, can you explain the process of the connect/disconnect and raising hand process by attendee?

**Comment by michhyun1** [comment]
The raise hand feature should not be correlated to the attendee video not showing. You may want to confirm that you are using the correct rtmpEndpoint when starting transcoding. And you should also make sure you've started the MediaLive channel - the ID is specified in the cloudformation outputs. In some cases, you may want to redeploy a new stack in case something was updated incorrectly.

**Comment by vijayamin83** [comment]
@michhyun1 Yes, I have found the flow of the raise hand. 

But I am still don't get the attendee video when I test transcoding lambda function. 

So, Let me explain you. I put my broadcast url in the test event and I also run the broad cast url in my local browser. So, is this something that will be broadcasted to RTMP endpoint? 

Or Should I have to use the OBS software to stream the RTMP EndPoint separately?

**Comment by vijayamin83** [comment]
I have confirmed that I am using the correct Media Live Channel ID. Also set the proper RTMP endpoint. But still getting the error.

**Comment by michhyun1** [comment]
@vijayamin83 Ah I see. You may not be seeing the video because you are joining with the broadcast URL. You shouldn't need to open the broadcast URL at all.

**Comment by vijayamin83** [comment]
@michhyun1 I am not opening the broadcast URL in my system or any where. But still I can't see the video on the Attendee Video.

Another question, If broadcast url open in the EC2 on the fly, then which video/audio it will be broadcast?

**Comment by michhyun1** [comment]
@vijayamin83 that depends on which URL you use when you start the transcoding lambda:
```
{
  "action": "start",
  "meetingURL": "<your-meeting-url-to-be-broadcast>", <-- use the broadcast meeting URL
  "rtmpEndpoint": "<your-rtmp-endpoint-url>" <-- this should be part of your outputs
}
```

The meeting URL field should be the broadcast URL that should be part of the CloudFormation Outputs when you deploy your live events stack. Similarly, the rtmp endpoint should be the MediaLivePrimaryEndpoint in your CloudFormation output

**Comment by michhyun1** [comment]
@vijayamin83 Can you please create a new issue for the attendee video not showing - it is unrelated to the raise hand feature

**Comment by vijayamin83** [comment]
creating the new ticket https://github.com/aws-samples/amazon-chime-live-events/issues/48
